[songgthu] iP

songgthu opened the pull request:
### MIL CHATBOT
> “Your mind is for having ideas, not holding them.” – David Allen

**Mil is a personal chatbot that helps you managing tasks** 😉
- User-friendly
- ~~FAST~~ SUPER FAST to use

**Here are the steps to start the chatbot:** 
1. Downoad the app [here](https://github.com/songgthu/ip)
2. Run Mil.java
3. Add and customize your task list
4. Type _**"bye"**_ to exit the chatbot 

**Current features:**
- [ x ] Managing different types of tasks (todo, deadline, event)
- [ x ] Save your task list after you exit the chatbot

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

songgthu committed:
Level 0

songgthu committed:
Level 1

songgthu committed:
Level 2

songgthu committed:
Level 3

songgthu committed:
Level 3

songgthu committed:
Level 3

songgthu committed:
Level 3

songgthu committed:
Merge branch 'master' of https://github.com/songgthu/ip

songgthu committed:
Level 4

songgthu committed:
Automated Text UI Testing

songgthu committed:
Level 5

songgthu committed:
Level 6

songgthu committed:
Level 6

songgthu committed:
Add save function

songgthu committed:
Add date formatting for Deadline, Event tasks

songgthu committed:
Merge pull request #1 from songgthu/branch-Level-7

Add date formatting for Deadline, Event tasks

songgthu committed:
Extract the following classes: Ui, Storage, Parser, TaskList

songgthu committed:
Organize classes into packages

songgthu committed:
Set up gradle

songgthu committed:
Add JUnit testcases for Todo, Deadline and Event

songgthu committed:
Add JUnit testcases of Todo, Deadline and Event

songgthu committed:
Package the app as a JAR file

songgthu committed:
Add javadoc to classes and methods

songgthu committed:
Check coding standards

songgthu committed:
Add find task function

songgthu committed:
Merge branch 'branch-A-CodingStandards'

songgthu committed:
Merge branch 'branch-Level-9'

songgthu committed:
Add checkStyle to detect coding style violations

songgthu committed:
Create a simple GUI for the app

A1WAYSD commented on src/main/java/milbot/Storage.java:
```diff
@@ -0,0 +1,61 @@
+package milbot;
+import taskclasses.*;
+
+import java.io.*;
```
Perhaps specifying the modules imported here to be more concise?

songgthu replied:
Thank you for pointing out, I will update it to be more precise!

A1WAYSD commented on src/main/java/milbot/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import taskclasses.*;
```
...I've noticed the same issue in other places too.

A1WAYSD commented on src/main/java/milbot/Ui.java:
```diff
@@ -0,0 +1,114 @@
+package milbot;
+
+import taskclasses.*;
```
...I noticed the same issue in other places too.

A1WAYSD commented on src/main/java/milbot/Ui.java:
```diff
@@ -0,0 +1,114 @@
+     * @param task     The added task.
+     */
+
+    public String printNewTask(TaskList taskList, Task task) {
+        String newTaskMessage = "Got it. I've added this task:\n" +
+                "  " + task + "\n" +
+                "Now you have " + taskList.getSize() + " tasks in the list.\n";
+
```
Perhaps when wrapping, try to break before an operator to improve readability?

songgthu replied:
Thanks for the suggestion!

A1WAYSD reviewed (commented):
Overall greatly done! Just some minor changes needed:)

Kailash201 commented on src/main/java/taskclasses/Event.java:
```diff
@@ -0,0 +1,27 @@
+
+public class Event extends Task {
+    protected String from;
+    protected String to;
+    public Event(String description, LocalDate from, LocalDate to) {
+        super(description);
```
Extremely minor issue with styling, I think it will make your code neater if there is a space between variables and method body as the rest of your files have space between them.

songgthu replied:
Thank you for your suggestion!

Kailash201 commented on src/main/resources/images/chatbot.png:
I like your chatbot image

Kailash201 commented on src/main/java/exception/NoTaskIndexException.java:
```diff
@@ -0,0 +1,7 @@
+    public NoTaskIndexException() {
+        super("☹ Oopsie! You did not input the task index.");
+    }
+}
```
I like how you have exception classes for each DukeException

Kailash201 commented on src/main/java/milbot/Mil.java:
```diff
@@ -0,0 +1,38 @@
+        parser = new Parser(taskList, ui, storage);
+    }
+
+//    public static void main(String[] args) {
+//        Scanner scanner = new Scanner(System.in);
+//        String input;
+//        //ui.printWelcomeMessage();
+//
+//        while (scanner.hasNext()) {
+//            input = scanner.nextLine();
+//            parser.parseInput(input);
+//        }
+//
+//    }
```
Will be neater if unwanted comments are removed from the codebase

songgthu replied:
Thanks for pointing it out!

Kailash201 reviewed (commented):
Overall, good attempt, and coding standards were well met. The use of OOP was good.
Just some minor issues with styling.
Good job!

songgthu committed:
Add assert statements to the code base

The current code lacks proper validation and error checking, making it susceptible to unexpected behavior and more difficult to debug.

In createTaskFromData method, `dataLine` is split into an array for parsing purpose, so it is necessary to make sure the array length does not exceed 4 to prevent parse issue.

Using assert statements is a best practice for improving code reliability and catch potential issues.

songgthu committed:
Improve code quality

In this commit, I refactored the code by breaking down a lengthy method into smaller, more manageable sub-methods. Restructuring the code not only enhances code readability but also reduces deep nesting. Additionally, I ensured consistent method naming throughout the codebase.

songgthu committed:
Merge pull request #2 from songgthu/branch-A-Assertions

Branch A-Assertions

songgthu committed:
Merge branch 'master' into branch-A-CodeQuality

songgthu committed:
Add tagging items extension

Functions that have been added:
- Create and delete tags
- Add tag to a task
- Remove tag from a task
- Auto-save tags

songgthu committed:
Fix minor bugs and improve Ui

songgthu committed:
Fix minor bugs and improve Ui

songgthu committed:
Add user guide

songgthu committed:
Update user guide

songgthu committed:
Update user guide

songgthu committed:
Finalize the produce

songgthu committed:
Finalize the product
